### PR TITLE
Upgrade Scala.js and Scala version patch levels then add Scala Native support

### PR DESCRIPTION
**Pull request by LeeTibbert**
We first align the Scala.js  & Scala version patch levels to those required by Scala Native
and then add Scala Native support.

### CODE REVIEW

File: .travis.yml
Comment (sjrd) [inline .travis.yml]: The empty `SCALAJS_VERSION` line should be removed. It is equivalent to `SCALAJS_VERSION=1.0.0`.

For the `SCALAJVM_VERSION` and `SCALANATIVE_VERSION`, could we set it to something non-empty, like `true` for example? That would be less awkward to read IMO.

File: build.sbt
Comment (sjrd) [inline build.sbt]: I believe keeping this one line is much more readable, especially because it's chained with other method calls:
```suggestion
lazy val `portable-scala-reflect` = crossProject(JSPlatform, JVMPlatform, NativePlatform)
```

File: build.sbt
Comment (sjrd) [inline build.sbt]: We use trailing commas in this build file:
```suggestion
    mimaPreviousArtifacts := Set.empty, // SN just added, no previous artifact
```

File: native/src/main/scala/org/portablescala/reflect/Reflect.scala
Comment (sjrd) [inline native/src/main/scala/org/portablescala/reflect/Reflect.scala]:
```suggestion
   *  In Scala Native, this method ignores the parameter `loader`. Calling this
```

File: native/src/main/scala/org/portablescala/reflect/Reflect.scala
Comment (sjrd) [inline native/src/main/scala/org/portablescala/reflect/Reflect.scala]:
```suggestion
   *  In Scala Native, this method ignores the parameter `loader`. Calling this
```

**Review by sjrd — CHANGES_REQUESTED** [review]
Thank you. The first two commits LGTM. I have a few comments on the third commit, which you can `git commit --amend` to integrate the fixes into the commit rather than make a new commit.

**Review by sjrd — APPROVED** [review]

**Comment by LeeTibbert** [comment]
Sjrd: Travis CI is all green!  Thank you for all the energy you invested.

The one small change after the Approval was to change "true" to true in two places 
in `.travis.xml`.   Although I like explicit strings, I think the latter reads better in the
log file.  I have been reading a lot of those recently.  I was not quick enough 
getting that change in and only saw your Approval after it had been committed.

What is the thinking about a release? I believe that takes some
Scala Center time for tags, artifacts, etc. 

 I can study the places where the version would need to change 
 to say `1.1.0` and create an Issue listing them (as checkboxes).
 My intent is to be useful, not to push the river.

**Comment by LeeTibbert** [comment]
@sideeffffect @lolgab Thank you for your contribution on bringing Scala Native support.  I think
this has been a team effort.  Of course, all remaining bugs, defects, etc. are on me.

**Comment by sjrd** [comment]
I will take care of the release later today/tomorrow (depending on your time zone).

**Comment by sjrd** [comment]
>  Of course, all remaining bugs, defects, etc. are on me.

Don't take too much credit. 😆 Remaining problems would be on me, since I approved all the code that went in.

**Pull request merged**
